tools/mason/MasonPublish.chpl

```diff
@@ -0,0 +1,128 @@
+/*
+*Design Plan
+*First fork the mason-registry repo
+*branch the repo and add the mason.toml into Bricks/<package-name>/<version>.toml
+*add source field to <version>.toml that points to the repo of the package
+*pull request from mason-registry from branch that contaons the <version>/toml
+*need to figure out how the environment of the package where mason is invoked passes its env to mason 
+*
+*
+*/
+
+use MasonUtils;
+use Spawn;
+use FileSystem;
+use TOML;
+use MasonEnv;
+use MasonNew;
+use MasonModify;
+
+proc masonPublish(args) throws {
+  /* Implement a try catch to catch the errors from functions within this function*/
+  try! {
+    if args.size > 2 {
+      for arg in args[2..] {
+        if arg == '-h' || arg == '--help'{
+          masonPublishHelp();
+          exit(0);
+        }
+      }
+    }
+    else {
+      if isGitExist() {
+        forkMasonReg();
```

Comment:
Many of these git commands rely on network calls that have the potential to fail. What is the error message when they fail?

You can force them to fail with a manual modification to the `runCommand` argument. e.g. `git clone https://github.com/this-does-not-exist/mason-registry mason-registry`